logic/pulsed/pulsed_measurement_logic.py

```diff
@@ -1142,29 +1157,47 @@ def _get_raw_data(self):
         """
         Get the raw count data from the fast counting hardware and perform sanity checks.
         Also add recalled raw data to the newly received data.
-        :return numpy.ndarray: The count data (1D for ungated, 2D for gated counter)
+        :return tuple(numpy.ndarray, info_dict): The count data (1D for ungated, 2D for gated counter) and
+                                                 info_dict with keys 'elapsed_sweeps' and 'elapsed_time'
         """
         # get raw data from fast counter
         fc_data = netobtain(self.fastcounter().get_data_trace())
+        if type(fc_data) == tuple and len(fc_data) == 2:  # if the hardware implement the new version of the interface
+            fc_data, info_dict = fc_data
+        else:
+            info_dict = {'elapsed_sweeps': None, 'elapsed_time': None}
+
+        if type(info_dict) == dict and 'elapsed_sweeps' in info_dict.keys() and info_dict['elapsed_sweeps'] is not None:
```

Comment:
Please avoid `type(some_var) == dict` or similar for these kind of checks.
It's better to use `isinstance(some_var, dict)` since this will also return `True` in case of derived types (e.g. `OrderedDict`).
You can also do something like `isinstance(some_var, (list, tuple))` to check for both.

Instead of 
`'elapsed_sweeps' in info_dict.keys() and info_dict['elapsed_sweeps'] is not None` 
you could do 
`info_dict.get('elapsed_sweeps') is not None`. This is more efficient, shorter and easier to read.